src/public/app/player/player.component.ts

```diff
@@ -25,20 +23,23 @@ const template: string = `
     template,
     selector: 'player',
 })
-export class Player implements OnChanges {
-
-    @Input() private song: Song;
+class Player implements OnChanges {
 
+    @Input() public queuedSong: QueuedUpSong;
+    @Output() public songFinished: EventEmitter = new EventEmitter();
     private player: ScPlayer;
```

Comment:
I'm not against that. We can probably use that to pre-fetch the players too and that way the player will just call `.play()`. Lets do that in another commit though.